rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/Serialization.kt

```diff
@@ -54,14 +56,44 @@ fun List<N_Lecture>.toJson() = LectureList(this.mapNotNull {
     }
 })
 
+/**
+ * A data class to represent the type of change that was sent/received by the websocket
+ */
 enum class WhatChanged {
     ROOM,
     ROOM_LIST,
     LECTURE,
     LECTURE_LIST
 }
-
+/**
+ * A data class to represent the change notification sent/received by the update websocket
+ */
 data class ChangeNotification(
-    val whatChanged: WhatChanged,
-    val change: Any
+        val whatChanged: WhatChanged,
+        val change:  Any
 )
+/**
+ * A Gson deserializer which can distinguish between the different WhatChanged types.
+ * Will return the deserialized ChangeNotification
+ */
+class ChangeNotificationDeserializer() : JsonDeserializer<ChangeNotification> {
```

Comment:
```suggestion
class ChangeNotificationDeserializer : JsonDeserializer<ChangeNotification> {
```
Remove empty primary constructor